plugins/discovery-azure/src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java

```diff
@@ -191,7 +186,7 @@ public AzureUnicastHostsProvider(Settings settings, AzureComputeService azureCom
             }
 
             // If provided, we check the deployment name
-            if (deploymentName != null && !deploymentName.equals(deployment.getName())) {
+            if (Strings.hasLength(deploymentName) && !deploymentName.equals(deployment.getName())) {
```

Comment:
I've done it that way so that we don't need to recompute the deploymentName / deploymentSlot all the time (as it's used in a loop, which is also called whenever we ping).